openpype/plugins/publish/collect_farm_target.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+import pyblish.api
+
+
+class CollectFarmTarget(pyblish.api.InstancePlugin):
+    """Collects the render target for the instance
+    """
+
+    order = pyblish.api.CollectorOrder + 0.499
+    label = "Collect Farm Target"
+    targets = ["local"]
+
+    def process(self, instance):
+        if not instance.data.get("farm"):
+            return
+
+        context = instance.context
+        try:
```

Comment:
So for someone to render on muster, they have to have both `deadline` and `royalrenderer` installed as modules?

Wouldn't an approach like this be better?
```py
farm_name = ""

op_modules = context.data.get("openPypeModules")

for farm_renderer in ["deadline","royalrender", "muster"]:
	op_module  = op_modules.get(farm_renderer, False)
	
	if op_module and op_module.enabled:
		farm_name = farm_renderer
	elif not op_module:
		self.log.error("Cannot get OpenPype {0} module.".format(farm_renderer))


if farm_name:
	self.log.debug("Collected render target: {0}".format(farm_name))
	instance.data["toBeRenderedOn"] = farm_name
```

Reply:
Yeah, definitely improvement. But also we are planning to remove `muster` from codebase.

Reply:
I used "muster" as an example, but the same applies to "royalrender", my point was that unless you have "deadline" module loaded, you won't be able to render in any of the other two

Reply:
I was thinking about exposing renderer as a dropdown in Publisher and filling it only with enabled modules. Not sure if it not a overkill if there is no use case for studio having multiple renderers active. (Maybe for different projects?)

Reply:
> I was thinking about exposing renderer as a dropdown in Publisher and filling it only with enabled modules. Not sure if it not a overkill if there is no use case for studio having multiple renderers active. (Maybe for different projects?)

It could be `hidden=True` if there's only one active?

Reply:
> I used "muster" as an example, but the same applies to "royalrender", my point was that unless you have "deadline" module loaded, you won't be able to render in any of the other two

Definitely! I haven't seen the raise after each try and it would be blocking. I had implemented our way and all works ;) Thanks